### PR TITLE
✨ [bento][npm] Update file naming from mjs to module

### PR DESCRIPTION
**Pull request by krdwan**
Update naming of generated files for NPM publishing.

Instead of using `.mjs`, we will go with `.module.js` since `create-react-app` does not support `.mjs` format.

### CODE REVIEW

File: build-system/tasks/helpers.js [hunk 1]
```diff
@@ -472,9 +480,9 @@ async function compileUnminifiedJs(srcDir, srcFilename, destDir, options) {
     watchedTargets.set(entryPoint, {
       rebuild: async () => {
         const time = Date.now();
-        const {rebuild} = /** @type {Required<esbuild.BuildResult>} */ (
-          buildResult
-        );
+        const {
+          rebuild,
+        } = /** @type {Required<esbuild.BuildResult>} */ (buildResult);
```
Comment (krdwan) [inline build-system/tasks/helpers.js:486]: This is unrelated.  Automatically updated by prettier.

**Review by jridgewell — APPROVED** [review]
Sweet!

**Comment by krdwan** [comment]
> Sweet!

Nice!!

File: build-system/tasks/helpers.js [hunk 2]
```diff
@@ -291,6 +291,14 @@ function maybeToEsmName(name) {
   return argv.esm ? toEsmName(name) : name;
 }
 
+/**
+ * @param {string} name
+ * @return {string}
+ */
+function npmMaybeToEsmName(name) {
```
Comment (caroqliu) [inline build-system/tasks/helpers.js:298]:
```suggestion
function maybeToNpmEsmName(name) {
```

File: build-system/tasks/extension-helpers.js [hunk 3]
```diff
@@ -615,8 +616,8 @@ function buildBinaries(extDir, binaries, options) {
       entryPoint,
       `${extDir}/dist`,
       Object.assign(options, {
-        toName: maybeToEsmName(`${name}.max.js`),
-        minifiedName: maybeToEsmName(`${name}.js`),
+        toName: npmMaybeToEsmName(`${name}.max.js`),
+        minifiedName: npmMaybeToEsmName(`${name}.js`),
```
Comment (caroqliu) [inline build-system/tasks/extension-helpers.js:620]: FMI: Why won't this give the npm naming to regular extension builds? It looks like `buildBinaries` is used here as well, where we would want the regular esm name:
https://github.com/ampproject/amphtml/blob/7aeb64c073d68050e5ca0ed9bd4e7820ed8737c2/build-system/tasks/extension-helpers.js#L491

Comment (krdwan) [reply build-system/tasks/extension-helpers.js]: `buildBinaries` is only used for generating NPM binaries

File: build-system/tasks/helpers.js (hunk 2 above)
Comment (krdwan) [reply build-system/tasks/helpers.js]: updated
